### PR TITLE
Fix MQTT handler error handling and API call issues

### PR DESCRIPTION
**Pull request by markus-lassfolk**
- Wrap MQTT message handler body in try/catch to prevent unhandled exceptions from breaking message processing
- Add null check for working_state before comparing to prevent false controller resets
- Wrap RefreshAuth() in try/catch within retry loop to allow retries to continue on refresh failures
- Fix DisconnectAsync calls to include required MqttClientDisconnectOptions parameter

## Summary

Closes #

## Type of change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [ ] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Refactor / code cleanup (no behaviour change)
- [ ] 📝 Documentation update
- [ ] ⚙️ CI/CD / tooling change

## Changes made

- 
- 

## Testing

- [ ] Added / updated Pester tests
- [ ] All tests pass locally (`Invoke-Pester ./tests`)
- [ ] Tested manually against Yarbo hardware (if applicable — describe setup)

## Quality checklist

- [ ] **PSScriptAnalyzer**: Zero Error or Warning findings (`Invoke-ScriptAnalyzer ./src -Settings ./PSScriptAnalyzerSettings.psd1 -Severity Error,Warning`)
- [ ] **Tests pass**: `Invoke-Pester ./tests` — all green
- [ ] **Module manifest updated** if version changed
- [ ] **CHANGELOG.md** updated under `[Unreleased]`
- [ ] **Docs updated** (comment-based help, README if needed)
- [ ] **No secrets, credentials, or PII** in code, comments, or commit messages
- [ ] **Follows coding standards** in [CONTRIBUTING.md](../CONTRIBUTING.md)

## Screenshots / output (optional)

---

> **Reviewer note**: Branch must be up-to-date with `develop` and all CI checks must pass before merge.

---

> [!NOTE]
> **Low Risk**
> Low risk: small, localized error-handling and API-signature fixes that mainly affect retry/disconnect behavior; main risk is masking underlying auth failures via warnings instead of surfacing immediately.
> 
> **Overview**
> Improves resilience of cloud API retries by wrapping the 401 token refresh path in `YarboCloudSession.InvokeWithRetry` with a `try/catch`, so GET retries can continue even if `RefreshAuth()` fails.
> 
> Fixes `Find-Yarbo` MQTT cleanup to call `DisconnectAsync` with the required `MqttClientDisconnectOptions` (built via the factory) instead of the previous signature, reducing disconnect-time errors during discovery.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a47aa3ee506588b8d16869121bd874fb53179540. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/PSYarbo/Public/Connection/Connect-Yarbo.ps1 [hunk 1]
```diff
@@ -134,36 +134,40 @@ function Connect-Yarbo {
...
                             }
                         }
-                    } elseif ($topic -like '*/device/DeviceMSG') {
-                        $decoded = ConvertFrom-ZlibPayload -Data $payload
-                        if ($decoded) {
-                            $conn.Robot = ConvertTo-YarboRobot -DeviceMsg $decoded -SerialNumber $conn.SerialNumber -Broker $conn.Broker -Port $conn.Port
-                        }
+                    } catch {
+                        Write-Debug "[Connect-Yarbo] MQTT message handler error (non-fatal): $($_.Exception.Message)"
                     }
```
[the remark this answers is not in the document]
Comment (markus-lassfolk) [reply src/PSYarbo/Public/Connection/Connect-Yarbo.ps1]: Test coverage noted for follow-up. The bug fix itself is correct and verified manually. Adding Pester tests for MQTT edge cases and auth retry is planned.
